File: frontend/src/dtos/SearchCriteria.ts
import PaginatedDTO from "./PaginatedDTO.ts";

export default interface SearchCriteria extends PaginatedDTO {
    title?: string;
    maxRuntime?: number;
    minRuntime?: number;
    genre?: string;
    year?: number;
    maxRating?: number;
    minRating?: number;
}

File: backend/src/main/java/com/ega/imdb/dtos/SearchCriteria.java
package com.ega.imdb.dtos;

import lombok.Builder;
import lombok.Data;

@Builder
@Data
public class SearchCriteria {

    private final String title;
    private final Integer maxRuntime;
    private final Integer minRuntime;
    private final Integer maxRating;
    private final Integer minRating;
    private final String genre;
    private final Integer year;

    private final int pageNumber;
    private final int pageSize;
    private final String sortingField;
    private final boolean ascendingSorting;
}
